elyra/util/cos.py

```diff
@@ -46,16 +46,23 @@ def __init__(self, config=None, endpoint=None, access_key=None, secret_key=None,
             self.bucket = config.metadata['cos_bucket']
             if auth_type in ['USER_CREDENTIALS', 'KUBERNETES_SECRET']:
                 # TODO remove temporary validation checks after
-                # the 'cos_username' and 'cos_password' schema properties
-                # are marked as required.
+                # metadata can properly enforce constraints
                 if len(config.metadata.get('cos_username', '').strip()) == 0 or\
                    len(config.metadata.get('cos_password', '').strip()) == 0:
-                    raise RuntimeError(f'Authentication provider \'{auth_type}\' requires '
+                    raise RuntimeError('Cannot connect to object storage. '
+                                       f'Authentication provider \'{auth_type}\' requires '
                                        'a username and password. Update runtime configuration '
                                        f'\'{config.display_name}\' and try again.')
+                if auth_type == 'USER_CREDENTIALS' and\
+                   len(config.metadata.get('cos_secret', '').strip()) > 0:
+                    raise RuntimeError('Cannot connect to object storage. '
+                                       f'Authentication provider \'{auth_type}\' does not '
+                                       'support Kubernetes secrets. Update runtime configuration '
+                                       f'\'{config.display_name}\' and try again.')
```

Comment:
Do we need to validate items that will be ignored anyway?  This will make it frustrating for users to switch between authentication types if they're forced to unset various fields depending on the type.  If this is required, I would suggest it too move in the appropriate metadata class hook method.

Reply:
I guess I don't agree here.  If I have `auth_type` == `KUBERNETES_SECRET` working but want to switch to `USER_CREDENTIALS` how do I unset `cos_secret` assuming the UI will hide that field when I've switched to `USER_CREDENTIALS`.   Must I unset the field _prior_ to switching `auth_type`?  If `cos_secret` is ignored for `USER_CREDENTIALS`, why enforce that it must have no value?

I think the same could be said about `AWS_IAM_ROLES_FOR_SERVICE_ACCOUNTS`.  When set, ignore the others.

Once we have support for `oneOf` and object-valued properties this form of validation won't be applicable since the fields will be a function of the `auth_type`.